app/public/js/location_report.js

```diff
@@ -14,6 +14,22 @@ function initMap() {
 	  map: map
 	});
 
+	lastRight = null;
```

Comment:
If lastRight, newRight don't have to be global variables, use "var" to declare them with scope local to the function instead.

This type of feedback applies to both location_report.js and location_search_map.js.

The larger the scope of any variable, the higher the risk of some other code clashing with it by reusing the name for something different.  For example, if you think that 'lastRight' is safe to use globally and someone else thinks the same thing, side effects from one piece of code will potentially break the other.